icing/icing_optimisationProofsScript.sml

```diff
@@ -603,11 +603,376 @@ Proof
   \\ fs[]
 QED
 
+fun rename_each [] = ALL_TAC
+| rename_each [first] = qmatch_asmsub_rename_tac first \\ qpat_x_assum first mp_tac \\ disch_then assume_tac
+| rename_each (first::rest) = qmatch_asmsub_rename_tac first \\ qpat_x_assum first mp_tac \\ rename_each rest \\ disch_then assume_tac
+
+
 Theorem fp_assoc_gen_correct:
   ∀ fpBop st1 st2 env e r.
     is_rewriteFPexp_correct [fp_assoc_gen fpBop] st1 st2 env e r
 Proof
-  cheat
+  rw[is_rewriteFPexp_correct_def]
+  \\ REVERSE (qspecl_then [‘e’, ‘fpBop’] strip_assume_tac fp_assoc_gen_cases)
+  >- (
+  fs[]
```

Comment:
Can you remove the line break after the `fs[]`? It is usually a good practice to put "conceptual steps" on a single line, but there is no need for `fs[]` to be "alone"

Reply:
Similarly for snippets like
    
    rveq
    \\ fs[]

and other below. They should be streamlined into a single line.